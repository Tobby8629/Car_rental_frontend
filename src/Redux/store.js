import { configureStore } from '@reduxjs/toolkit';
import UserSlice from './UserSlice';
import CarSlice from './CarSlice';
import ReservationSlice from './ReservationSlice';

const store = configureStore({
  reducer: {
    Users: UserSlice,
    Cars: CarSlice,
    Reserve: ReservationSlice,
  },
});

export default store;
