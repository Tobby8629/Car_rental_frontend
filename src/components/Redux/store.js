import { configureStore } from '@reduxjs/toolkit';
import UserSlice from './UserSlice';
import ReservationSlice from './ReservationSlice';

const store = configureStore({
  reducer: {
    Users: UserSlice,
    Reserve: ReservationSlice,
  },
});

export default store;
