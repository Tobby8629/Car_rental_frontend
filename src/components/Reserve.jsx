import React, { useState } from 'react';
import { useSelector } from 'react-redux';
import './componentsCss/reservation.css';

function Reservation() {
  // const cars = useSelector((state)=> state.Cars.cars)
  const storage = localStorage.getItem('token');
  const user = JSON.parse(storage);
  const [reserve, setreserve] = useState({
    userId: user.id, city: '', pickup: '', date: '', returnDate: '', carId: '',
  });

  const submit = async (e) => {
    e.preventDefault();
    const {
      city, pickup, date, returnDate, carId,
    } = reserve;

    // if (!city || !pickup || !date || !returnDate || !carId) {

    // }

    // dispatch(postUser({
    //   reserve: {
    //     user: reserve.userId, city: reserve.city, pickup: reserve.pickup,
    //    date: reserve.date, car: reserve.carId, return: reserve.return_date,
    //   },
    // }));
  };
  return (
    <section>
      <div className="reservation form">
        <form onSubmit={submit}>
          <h2>Make reservation with us</h2>
          <input type="text" id="city" value={reserve.city} placeholder="city" onChange={(e) => setreserve({ ...reserve, city: e.target.value })} />
          <input type="text" id="pickup" value={reserve.pickup} placeholder="pickup" onChange={(e) => setreserve({ ...reserve, pickup: e.target.value })} />
          <input type="date" id="date" value={reserve.date} placeholder="date" onChange={(e) => setreserve({ ...reserve, date: e.target.value })} />
          <input type="date" id="date" value={reserve.return_date} placeholder="date" onChange={(e) => setreserve({ ...reserve, return_date: e.target.value })} />
          <div className="form-btn">
            <button type="submit">Reserve</button>
          </div>
        </form>
      </div>
    </section>
  );
}

export default Reservation;
