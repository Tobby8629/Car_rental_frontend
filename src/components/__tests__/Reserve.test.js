import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../../Redux/store';
import Reserve from '../Reserve';

test('Reserve component matches snapshot', () => {
  const user = { id: '123' }; // Set the desired user ID here
  const localStorageMock = {
    getItem: jest.fn().mockReturnValue(JSON.stringify(user)),
  };
  Object.defineProperty(window, 'localStorage', {
    value: localStorageMock,
    writable: true,
  });

  const { asFragment } = render(
    <Provider store={store}>
      <BrowserRouter>
        <Reserve />
      </BrowserRouter>
    </Provider>,

  );
  expect(asFragment()).toMatchSnapshot();
});
