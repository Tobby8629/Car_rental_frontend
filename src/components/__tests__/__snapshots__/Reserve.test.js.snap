// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reserve component matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="rees"
  >
    <div
      class="reservation form"
    >
      <form>
        <h2>
          Make reservation with us
        </h2>
        <div
          class="redone"
        >
           All field must be filled
        </div>
        <input
          id="city"
          placeholder="city"
          type="text"
          value=""
        />
        <input
          id="pickup"
          placeholder="pickup"
          type="text"
          value=""
        />
        <input
          id="date"
          placeholder="date"
          type="date"
          value=""
        />
        <input
          id="date"
          placeholder="date"
          type="date"
          value=""
        />
        <div
          class="red"
        >
          your return must not be before your pickup date
        </div>
        <select>
          <option
            value=""
          >
            select
          </option>
        </select>
        <div
          class="form-btn"
        >
          <button
            type="submit"
          >
            Reserve
          </button>
        </div>
      </form>
    </div>
  </section>
</DocumentFragment>
`;
